### PR TITLE
DOC: Make _field_accessor manage the docstring format

### LINKED ISSUE #24068: DOC: Make _field_accessor manage the docstring format
We started to validate the formats of docstrings. For example, we expect that the summary of the docstring starts in the next line of the opening `"""` (see #24059).

In some cases the docstrings are generated dynamically, and in the case of `_field_accessor`, we have to write some line breaks to avoid the validation to fail, for example:
```
_field_accessor('year', 'Y', "\n The year of the datetime\n")
```

In this issue we should update the code of `_field_accessor`, and modify the calls to it removing the line breaks, so the previous example becomes:
```
_field_accessor('year', 'Y', 'The year of the datetime')
```

And the docstring is composed in a way that follows our docstring standards. The standards can be checked with:
```
$ ./scripts/validate_docstrings.py --prefix=pandas.Series --errors=GL01,GL02
```
which, it should report no error (or if it reports cases that are not fixed yet, the number of errors should decrease or not increase after the change).

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #24068 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -83,6 +83,10 @@ def f(self):
         return result
 
     f.__name__ = name
+    if not docstring.startswith("\n"):
+        docstring = "".join(("\n", docstring))
+    if not docstring.endswith("\n"):
+        docstring = "".join((docstring, "\n"))
     f.__doc__ = docstring
```
Comment (datapythonista) [inline pandas/core/arrays/datetimes.py]: I'd make it as simple as:
```suggestion
f.__doc__ = '\n{}\n'.format(docstring)
```
Without the code you added.

Then, in the calls to `_field_accessor` we need to remove the `\n` and spaces around the description. I think that should work in all cases.

**Review by datapythonista — CHANGES_REQUESTED** [review]
I think we can make this simpler, see comment.

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (charlesdong1991) [reply pandas/core/arrays/datetimes.py]: oh, okay, i did think about this, then i thought if in the call to `_field_accessor`, we already used \n, then directly adding \n will be redundant... so i made a `if` judgement first.. 
but agree that if remove the `\n` added before, then this should not be the problem and look much simpler...

Comment (datapythonista) [reply pandas/core/arrays/datetimes.py]: we may have to do what you did if at some point we have something like:
```
_doc = """
Some summary.
"""
foo = _field_accessor(..., _doc)
```
But we'll complicate things at that point, if they ever happen. And to join two strings, I think it's better to use `'\n' + foo` of `'\n{}'.format(foo)` than using a join.

Comment (charlesdong1991) [reply pandas/core/arrays/datetimes.py]: okay, thanks for the tip!! @datapythonista

**Pull request closed without merging**